[WIP] [EPIC] MINT-1824: To complete refactoring

ozalexo opened the pull request.

ozalexo committed:
Removed console.log(s).

ozalexo committed:
MINT-1805: WIP

ozalexo committed:
MINT-1805: WIP

ozalexo committed:
MINT-1805: WIP
Patched redux-axios-middleware to obtain selectied network from Redux
Accumulated HTTP API (new @chronobank/api package)
Working on CommonNetworkSelector

ozalexo committed:
MINT-1805: Reworked CommonNetworkSelector component
New redux reducer NAVSET: now we can just switch networks and read current network from store
Working on HTTP API and 'provider' reconnect

ozalexo committed:
MINT-1777: WIP. Web3 middleware
Web3 now available before login
Web3 provider may be changed on the fly
Network structure reworked
Not done: need additional fixes with all methods, which using olf 'network' state

ozalexo committed:
Store and Router upgrade and fix

ozalexo committed:
WIP

ozalexo committed:
It works

ozalexo committed:
WIP

ozalexo committed:
Coorect initial Nav. NavigateTo still does not work. Something wrong with resux store 'router' - it is Immutable Map, need to work correctly eith that.

ozalexo committed:
Removed react-router-redux package (not used anymore)

ozalexo committed:
Fixed routing

ozalexo committed:
Removed a lot of binds, some additional fixes for routs

ozalexo committed:
WIP: removed some unnecessary code. Fixes in LoginForm

ozalexo committed:
WIP: updated pacakges, rebased on signers branch, fixes login forms, made redux-based authentication. Next: networking.

ozalexo committed:
Initialization after rehydrate made correctly. Removed LocalStorage. Fixed locale and network init

ozalexo committed:
Removed/replaced HTTP provider PublicBackendProvider

ozalexo committed:
Chronobank middlewares API. Blockdozer API

e-maximus commented on package.json:
```diff
@@ -120,51 +151,23 @@
-    "redux-immutable": "^3.0.10",
-    "redux-persist-immutable": "^4.3.1",
+    "redux-immutable": "^3.0.11",
+    "redux-persist-immutable": "hoavo/redux-persist-immutable#3d54b8f7f67f26bc2fd056c2d5b01401af23a149",
```
Is it stable for release to use this commit?

e-maximus commented on packages/core/redux/bitcoin/thunks.js:
```diff
@@ -123,13 +123,24 @@ export const getTransactionsList = (address: string, id, skip = 0, offset = 0, b
+  //     },
+  //   },
+  // })
+
```
Do we need this comment?

e-maximus commented on packages/core/redux/ducks.js:
```diff
@@ -18,15 +16,17 @@ import market from './market/reducer'
 import notifier from './notifier/reducer'
-import persistAccount from './persistAccount/reducer'
+// persistAccount reduxer configuring separately in configureStore for peristance
+// import persistAccount from './persistAccount/reducer'
```
Comments?

e-maximus commented on packages/core/redux/persistAccount/reducer.js:
```diff
@@ -3,83 +3,98 @@
+        payload.wallet,
+      ],
+    }
+  },
```
Why do we add wallets in persistAccount reducer? What do you mean wallet in this way?

e-maximus commented on packages/core/redux/persistAccount/selectors.js:
```diff
@@ -46,21 +46,22 @@ export const getEthereumSigner = (state) => {
-    }
+    // if (!network || !network.selectedNetworkId || !network.networks) {
+    //   return null
+    // }
```
?

e-maximus commented on packages/core/services/WebSocketService.js:
```diff
@@ -0,0 +1,200 @@
+            },
+          },
+        })
+        WebSocketService.reconnectByHeartBeatTimeout(wsInternalKey, 120000)
```
It's better to use constant for 12000

e-maximus commented on packages/nodes/httpNodes/api/bitcoin_middlewares.js:
```diff
@@ -0,0 +1,138 @@
+      return result
+    })
+    .catch((error) => {
+      if (blockchain === BLOCKCHAIN_BITCOIN_CASH) {
```
Why do we use BCC only here?

e-maximus commented on packages/nodes/redux/nodesMiddleware.js:
```diff
@@ -0,0 +1,149 @@
+let syncIntervalId = null // Using to check if node is syncing
+
+// TODO: Need to clarify algorythm and what to do case of errors
+const checkIfSyncing = (dispatch) => {
```
There are couple of oral question I have about these nodes. Ask me when you will be here.

e-maximus commented on packages/login-ui/redux/thunks.js:
```diff
@@ -114,7 +98,8 @@ export const onSubmitLoginForm = (password) => async (dispatch, getState) => {
         const providerSettings = dispatch(SessionThunks.getProviderSettings())
-        const provider = privateKeyProvider.getPrivateKeyProvider(wallet.privateKey.slice(2, 66), providerSettings)
+        const networkType = selectCurrentNetworkType(state)
+        const provider = privateKeyProvider.getPrivateKeyProvider(wallet.privateKey.slice(2, 66), { ...providerSettings, networkType })
```
The main goal is to get rid of privateKeyProvider. Dont forget it my friend

e-maximus reviewed (changes requested):
Certain questions is going to be discussed orally

olegdizus committed:
MINT-1846 Fix load translations (#7)

- Use correct translations keys
- Move loadI18n function to thunks
- Simplify translations processing flow
- Increase readability
- Extract isEntryNotEmptyObject predicate to utils
- Remove dead code

ozalexo committed:
Fixed unused variable

ozalexo committed:
WIP

e-maximus committed:
Merge branch 'feature/MINT-1752-refactor-send-tokens-form' into mergeForDev

olegdizus committed:
MINT-1876 Fix and refactor auth routing (#8)

* MINT-1876 Add isSession selector

* MINT-1876 Ignore jsconfig

* MINT-1876 Add protected routes

* MINT-1901 Simplify ProtectedRoute

* MINT-1904 Null-check privateKey in PKIcon

* MINT-1904 Fix selector name

* MINT-1904 Rework route layout usage
